x-pack/elastic-agent/pkg/agent/application/reexec/manager.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package reexec
+
+import (
+	"sync"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+)
+
+var (
+	execSingleton     ExecManager
+	execSingletonOnce sync.Once
+)
+
+// ExecManager is the interface that the global reexec manager implements.
+type ExecManager interface {
+	// ReExec asynchronously re-executes command in the same PID and memory address
+	// as the currently running application.
+	ReExec()
+
+	// ShutdownChan returns the shutdown channel the main function should use to
+	// handle shutdown of the current running application.
+	ShutdownChan() chan bool
```

Comment:
can we make it receive only channel? or do we intend to allow closing(triggering) outside of manager?
if so we might panic on Reexec and closing closed channel

Reply:
Good catch, yes it should be receive only.